### DISCUSSION TITLE
Substract Rows in Table

### DISCUSSION THREAD

**Discussion by muksidin** [Budibase/budibase#11733] [Help] [answered]
Hi.. is it possible to subsctract row in one column and the result in next column. I think you understand what i mean because it is like the spreadsheet basic formula.
[
![Screenshot_20230910_103400](https://github.com/Budibase/budibase/assets/5665115/0de9141a-2460-4c58-8f1b-c42da6663785)
](url)

**Comment by ConorWebb96** [comment]
Hey @muksidin,

Are you asking how to do this within Budibase?

1. The first step would be to make sure that both columns are number types. Otherwise you will have to convert them from string to numbers.
2. Create a formula column. Change it from dynamic to static.
<img width="100%" alt="Screenshot 2023-09-11 at 08 29 59" src="https://github.com/Budibase/budibase/assets/126772285/e91b259e-e729-4702-8990-75ac8c4e8626">
3. Click the lighting bolt, use the subtract filter and add your 2 columns you wish to run the calculation on.
<img width="100%" alt="Screenshot 2023-09-11 at 08 30 08" src="https://github.com/Budibase/budibase/assets/126772285/07deadaf-9201-44e3-87d9-366fa5343a68">
4. Click save and it should do run the fucntion and output the result within the formula column.

**Output**
<img width="676" alt="Screenshot 2023-09-11 at 08 29 48" src="https://github.com/Budibase/budibase/assets/126772285/4a5d735e-1e3a-4946-a1b2-12b54daa16b1">

I hope this helps.

**Reply by muksidin** [reply]
@ConorWebb96 , thanks for your answer. But my mean is how to substract row in one column. So in my screenshoot, column CONCUMPTION is result of substract row after and before in column METER. For example, the result 300 in column CONCUMPTION is substract 20600 - 20300.

**Reply by ConorWebb96** [reply]
I'm not sure what you are asking here? The example I gave will work for subtracting -  20600 - 20300.

**Reply by muksidin** [reply]
Ouh sorry for that. 
it should be the title **Substract Row in Column** not **Table**. So for example, 20600 - 20300 = 300, 21000 - 20600 = 400, etc.

**Reply by muksidin** [reply]
@ConorWebb96 , my mean is how to get difference value between 2 rows in column? .Like in my question picture.
Thanks

**Reply by ConorWebb96** [reply]
I think I understand what you are asking now. I don't believe this is possible as within the formula field you don't have access to any of the other rows within the table, its purely specific to its own row.

You could do this within the design area as you would have access to all rows within the table. From here you could then post the results back to the table. Doing it this way the concumption field would be set to a number instead of a formula.

Additionally you could use a cron job automation to fully automate this process for you. Pass in the 2 rows, do some calculations and then update the concumption field of the specific row with the result.

I hope this helps.

**Reply by muksidin** [reply]
This's difficult enough for me, i can not more to type in javascript. Can you help me to do it? I need that to calculation for our electrical consumption.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @muksidin,

Heres an app demo of how this could be done with an automation. If you delete records make sure to change the `endArea` boolean checkbox to ticked as it does use whatever the last row is whenever performing the automation.
[automation-concumption-1695625214713.tar.gz](https://github.com/Budibase/budibase/files/12712084/0.Subtract-export-1695625214713.tar.gz)

I hope this helps.

**Reply by muksidin** [reply]
Thanks alot @ConorWebb96 for the answer, i try to learn more about that.
